ProfileView/src/index.ts

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright (c) Cisco Systems, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+// This file imports all of the webcomponents from "components" folder
+
+import {
+  html,
+  internalProperty,
+  property,
+  customElement,
+  LitElement
+} from "lit-element";
+// import "./components/ActivityStream/ActivityItem";
+// import "./components/View/View";
+import "@cjaas/common-components";
+import { Profile } from "./types/cjaas";
+import styles from "./assets/styles/View.scss";
+
+/**
+ * Please give your widget a unique name. We recommend using prefix to identify the author and help avoid naming conflict. e.g. "2ring-timer-widget"
+ */
+@customElement("cjaas-profile-view-widget")
+export default class CjaasProfileWidget extends LitElement {
+  @property() customer: string | undefined;
+  @property() template: any | null | undefined = null;
+  @property({ attribute: "auth-token" }) authToken:
+    | string
+    | null
+    | undefined = null;
+
+  @property({ type: String, attribute: "base-url" }) baseURL =
+    "https://trycjaas.exp.bz";
+
+  @internalProperty() profile: any;
+  @internalProperty() presetTags: any = {};
+  @internalProperty() showSpinner = false;
+
+  updated(changedProperties: any) {
+    changedProperties.forEach((oldValue: string, name: string) => {
+      console.log(oldValue);
+      if (
+        (name === "template" || name === "customer") &&
+        this.customer &&
+        this.template
+      ) {
+        this.getProfile();
+      }
+    });
+  }
+
+  getProfile() {
+    const url = `${this.baseURL}/profileview?personid=${this.customer}`;
+    this.showSpinner = true;
+    this.requestUpdate();
+
+    // set verbose as true for tabbed attributes
+    const template = Object.assign({}, this.template);
+    template.Attributes = template.Attributes.map((x: any) => {
+      if (x.type === "tab") {
+        x.Verbose = true;
+      }
+      return x;
+    });
+
+    const body = JSON.stringify(template);
+
+    return fetch(url, {
```

Comment:
how do you like fetch api? :) feel free to use axios btw if you'd like, it has better browser compatibility and a bit less verbose. but goot to practice fetch also

Reply:
Ah good to know. Fetch is fine, but I'm all bout better compatability